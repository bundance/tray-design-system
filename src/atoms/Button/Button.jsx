import React from "react";
import {
  background,
  borderRadius,
  color,
  fontFamily,
  fontSize,
  fontWeight,
  layout,
  lineHeight,
  space,
} from "styled-system";
import styled from "styled-components";

const Button = styled.button(
  background,
  borderRadius,
  color,
  fontFamily,
  fontSize,
  fontWeight,
  layout,
  lineHeight,
  space
);

Button.propTypes = {
  ...background.propTypes,
  ...borderRadius.propTypes,
  ...color.propTypes,
  ...fontFamily.propTypes,
  ...fontSize.propTypes,
  ...fontWeight.propTypes,
  ...layout.propTypes,
  ...lineHeight.propTypes,
  ...space.propTypes,
};

Button.defaultProps = {
  color: "white",
  background: "buttonBlue",
  borderRadius: 3,
  fontFamily: "Nunito Sans",
  fontStyle: "normal",
  fontSize: 6,
  fontWeight: 3,
  height: 40,
  paddingX: 16,
};

export default Button;
