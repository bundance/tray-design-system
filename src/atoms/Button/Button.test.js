import React from "react";

import { render } from "@testing-library/react";
import Button from "./Button";

test("it renders correctlyy", () => {
  const { container } = render(<Button />);
  expect(container.firstChild).toMatchSnapshot();
});
