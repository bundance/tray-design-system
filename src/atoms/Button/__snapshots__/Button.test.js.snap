// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctlyy 1`] = `
<button
  class="sc-AxjAm bjOgFU"
  color="white"
  font-family="Nunito Sans"
  font-size="6"
  font-style="normal"
  font-weight="3"
  height="40"
/>
`;
